ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -705,27 +798,27 @@ void updateFilter(final Predicate<Packet> filter) {
       this.filter = filter;
     }
 
-    /**
-     * Executes the action associated with this state. Sets a "boomerang" timer to itself in case
-     * the action is retryable.
-     *
-     * @param lastTimeout the previous timeout, or 0 if this is the first time the action is being
-     *     executed.
-     */
-    void execute(final long lastTimeout, final int retryCount) {
+    /** Executes the action associated with this state. Sets a "boomerang" timer to itself. */
+    void execute() {
       action.accept(this);
-      if (retryable && retryCount < MAX_RETRIES) {
-        final long newTimeout = retryDelayFunction.apply(lastTimeout);
+      if (retryCount < MAX_RETRIES) {
+        delay = retryDelayFunction.apply(delay);
```

Comment:
If `delay` is set to `0`, does this just default to 2000ms delay every time?

Reply:
delay is a field (I changed the code here to this.delay to make that clearer) and as such the delay will be updated every time the execute is called.